pkg/graph/builder.go

```diff
@@ -250,21 +250,40 @@ func (b *Builder) NewResourceGraphDefinition(originalCR *v1alpha1.ResourceGraphD
 	return resourceGraphDefinition, nil
 }
 
+func (b *Builder) buildExternalRefResource(externalRef *v1alpha1.ExternalRef, resourceObject *map[string]interface{}) {
+	(*resourceObject)["apiVersion"] = externalRef.APIVersion
+	(*resourceObject)["kind"] = externalRef.Kind
+	metadata := map[string]interface{}{
+		"name": externalRef.Name,
+	}
+	if externalRef.Namespace != "" {
+		metadata["namespace"] = externalRef.Namespace
+	}
+	(*resourceObject)["metadata"] = metadata
+}
+
 // buildRGResource builds a resource from the given resource definition.
 // It provides a high-level understanding of the resource, by extracting the
 // OpenAPI schema, emulating the resource and extracting the cel expressions
 // from the schema.
-func (b *Builder) buildRGResource(rgResource *v1alpha1.Resource, namespacedResources map[k8sschema.GroupKind]bool, order int) (*Resource, error) {
+func (b *Builder) buildRGResource(rgResource *v1alpha1.Resource,
+	namespacedResources map[k8sschema.GroupKind]bool, order int) (*Resource, error) {
```

Comment:
nit (just keep the style consistent with the rest of the code base):
```suggestion
func (b *Builder) buildRGResource(
    rgResource *v1alpha1.Resource,
	namespacedResources map[k8sschema.GroupKind]bool,
	order int,
) (*Resource, error) {
```